src/FsAutoComplete.Core/Commands.fs

```
@@ -1139,23 +1155,55 @@ type Commands (checker: FSharpCompilerServiceChecker, state: State, backgroundSe
-          let checker : FSharpChecker = checker.GetFSharpChecker()
-          // ENHANCEMENT: consider caching the Fantomas configuration and reevaluate when the configuration file changes.
-          let config =
-              match Fantomas.Extras.EditorConfig.tryReadConfiguration (UMX.untag file) with
-              | Some c -> c
-              | None ->
-                fantomasLogger.warn (Log.setMessage "No fantomas configuration found for file '{filePath}' or parent directories. Using the default configuration." >> Log.addContextDestructured "filePath" file)
-                Fantomas.FormatConfig.FormatConfig.Default
-          try
-            let! formatted = Fantomas.CodeFormatter.FormatDocumentAsync(UMX.untag file, Fantomas.SourceOrigin.SourceText text, config, parsingOptions, checker)
-            return text, formatted
-          with
-          | ex ->
-            fantomasLogger.warn (Log.setMessage "Errors while formatting file, defaulting to previous content. Error message was {message}" >> Log.addContextDestructured "message" ex.Message >> Log.addExn ex)
-            return! Core.Error ex.Message
+        try
+          match fantomasDaemon.Value with
+          | None ->
+            fantomasLogger.warn (Log.setMessage "Fantomas daemon is not present")
+            return! Core.Error "Fantomas daemon is not present"
+          | Some fantomasService ->
+            let! _, text = x.TryGetFileCheckerOptionsWithLines file
+            let currentCode = string text
+            let filePath = (UMX.untag file)
+            let isLastFile =
+              state.GetProjectOptions file
+              |> Option.bind (fun options ->
+                fantomasLogger.debug (Log.setMessage (sprintf "Project options were found for \"%A\"" file))
+                Seq.tryLast options.SourceFiles)
+              |> Option.map (fun  lastFile -> lastFile = filePath)
+              |> Option.defaultValue false // TODO: not sure if this is a good default
```

@baronfel how unlikely is it that the project options are not found for a file?
The boolean value I'm after is that the parser should raise an error if there is no module or namespace.
Only scripts and the last file in a project are allowed to have this.

---

it should be very, very unlikely. I think in the case you can't find project options we should return a None result from this endpoint, similar to the 'no change' result mentioned above.